src/main/java/io/strimzi/utils/KafkaVersionService.java

```diff
@@ -0,0 +1,238 @@
+     * It's the released test-container-images image with the highest Kafka version number.
+     * The result will change when a new version of Kafka is released with a higher Kafka version number,
+     * and a test-container-images image which contains that Kafka version has been built and released.
+     * @return LogicalKafkaVersion latest release
+     */
+    public KafkaVersion latestRelease() {
+        // at least one release in the json schema is needed
+        if (this.logicalKafkaVersionEntities == null || this.logicalKafkaVersionEntities.size() < 1) {
+            throw new IllegalStateException("Wrong json schema! It must have at least one release");
+        }
+
+        KafkaVersion latestRelease = this.logicalKafkaVersionEntities.get(0);
+
+        for (int i = 1; i < this.logicalKafkaVersionEntities.size(); i++) {
+            if (latestRelease.compareTo(this.logicalKafkaVersionEntities.get(i)) < 0) {
+                latestRelease = this.logicalKafkaVersionEntities.get(i);
+            }
+        }
+
+        LOGGER.info("Latest release of Kafka is:{}", latestRelease.toString());
+
+        return latestRelease;
+    }
+
+    /**
+     * Gets the previous minor release.
+     * If this instance were 2.8.1 then at a certain point in time this method
+     * might return a version reflecting 2.7.2.
+     * On a later date it might return 2.7.3.
+     * It is allowed to cross major version boundaries.
+     * E.g. the previous minor release to 3.0.0 might be 2.8.2
+     * ============================================================
+     * | (prev prev minor) ---  (previous minor) ---  (current) |
+     *                                                     |
+     *                                                     *
+     * |    2.8.1   &lt;---&gt;     2.8.2  &lt;---&gt;   3.0.0  |
+     * ============================================================
+     * Assuming that test container images `kafka_versions.yaml` has following content:
+     * {
+     *   "version": 1,
+     *   "kafkaVersions": {
+     *      2.8.1": "test-container:latest-kafka-2.8.1",
+     *     "2.8.2": "test-container:latest-kafka-2.8.2",
+     *     "3.0.0": "test-container:latest-kafka-3.0.0"
+     *   }
+     * }
+     * @return LogicalKafkaVersion the previous minor release
+     */
+    public KafkaVersion previousMinor() {
+        if (this.logicalKafkaVersionEntities == null || this.logicalKafkaVersionEntities.size() < 1) {
+            throw new IllegalStateException("Wrong json schema! It must have at least one release");
+        }
+
+        // sort in this case always order the latest release at the end so the previous one is the previous minor release
+        this.logicalKafkaVersionEntities.sort(KafkaVersion::compareTo);
+
+        // 1.0.0
+        // 2.40.50
+        // 3.0.0 <- previous minor (it will be always previous last)
+        // 4.1.2 <- latest release
```

Comment:
Is this some leftover which should be deleted?

Reply:
we can remove it :) thanks